File: frontend/types/feedback-entry.ts
export interface FeedbackEntry {
  id: string;
  thread_id: string;
  author_name: string;
  author_role?: string | null;
  message: string;
  is_change_request: boolean;
  created_at: string;
}

export interface FeedbackEntryListResponse {
  data: FeedbackEntry[];
  meta: { count: number };
  error: null;
}

File: backend/app/schemas/feedback_entry.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class FeedbackEntryBase(BaseModel):
    thread_id: UUID
    author_name: str
    author_role: str | None = None
    message: str
    is_change_request: bool = False


class FeedbackEntryCreate(FeedbackEntryBase):
    pass


class FeedbackEntryRead(FeedbackEntryBase):
    id: UUID
    created_at: datetime


class FeedbackEntryListResponse(BaseModel):
    data: list[FeedbackEntryRead]
    meta: dict[str, int]
    error: None = None
